File: 프로그래머스/1/12926. 시저 암호/시저 암호.cs
using System;
public class Solution 
{
    public string solution(string s, int n) 
    {
        char[] arr = s.ToCharArray();
        for(int i = 0; i < arr.Length; i++)
        {
            if(arr[i] == ' ')
            {
                continue;
            }
            
            char baseChar = char.IsUpper(arr[i]) ? 'A' : 'a';
            int offset = ((arr[i] - baseChar) + n) % 26;
            arr[i] = (char)(baseChar + offset);
        }
        
        return new string(arr);
    }
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(string s, int n) 
{
    for(char &c : s)
    {
        if(c != ' ')
        {
            if (c >= 97 && c <= 122) 
            {
                if (c + n > 122) 
                {
                    c = c + n - 26;
                } 
                else 
                {
                    c = c + n;
                }
            }
            else if(c >= 65 && c <= 90)
            {
                if (c + n > 90) 
                {
                    c = c + n - 26;
                } 
                else 
                {
                    c = c + n;
                }
            }
        }
    }
    return s;
}